### PR TITLE
🌱  Update main branch book link

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: https://github.com/kubernetes/k8s.io/pull/2834#issuecomment-928183604

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
/assign @sbueringer 

working with netlify support to get the cert updated

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ avoid common [go style mistakes](https://github.com/golang/go/wiki/CodeReviewCom
 The documentation is published in form of a book at:
 
 - [Stable release](https://cluster-api.sigs.k8s.io)
-- [Tip of the main branch](https://tip.cluster-api.sigs.k8s.io/)
+- [Tip of the main branch](https://main.cluster-api.sigs.k8s.io/)
```
Comment (fabriziopandini) [inline CONTRIBUTING.md:65]: Q: should we add link to release banch docs (https://release-0-3.cluster-api.sigs.k8s.io, https://release-0-4.cluster-api.sigs.k8s.io)?

Comment (CecileRobertMichon) [reply CONTRIBUTING.md:65]: yes, I can do that in this PR as well

Comment (CecileRobertMichon) [reply CONTRIBUTING.md:65]: done

**Review by vincepri — APPROVED** [review]
/approve
/lgtm

**Pull request merged**